### DISCUSSION TITLE
Question about test data

### DISCUSSION THREAD

**Discussion by hongyingyue** [PaddlePaddle/PaddleSpatial#90] [BaiduKDDCup2022] [answered]
Dear host,

Could you please explain more detailed about test data sampling?

It's said that " we randomly sample a stride time step from the range [10min, 100min]. in other word, the stride time step are randomly ranged from 1h to 6h"

Well, I can understand both of sentence, but I can't connect the [10min, 100min] sampling, and 1h to 6h. 
- if sampling from 10min -100min, why they are ranged from 1h to 6h
- another question is that, all phase test data is within 15 days, am I understand right?

Thanks in advance,
Best regards

**Comment by ramber1836** [comment, marked as answer]
Hi, there,

The second sentence is incorrect. Apologies for that.
For your second question, the current phase test data is not within 15 days
and the exact time span is not provided. But one thing for sure is that the
test samples are chronologically ordered.
Thanks for your support.

Regards,
Xinjiang Lu

**Comment by hongyingyue** [comment]
@ramber1836 

Thanks, I understand.
For chronologically ordered test samples, do you mean the test samples dates are after train samples? or you mean the test samples in test folder are ordered ?

**Comment by ramber1836** [comment]
The test samples in the test folder are ordered.

**Comment by hongyingyue** [comment]
@ramber1836 

Thanks. But I don't fully understand why the test folder need orders here. My current understanding is,  just use the specific one csv from test_x to generate one example's feature, and generate the same name's file in test_y for evaluation. Then loop it.

If it's ordered, so we can use maybe more test_x files to generate one example? Is this the purpose of test folder ordering? 

Actually, the order also adds the risk of leaking here. If it's ordered, maybe we can use the future test_x to update the previous prediction if the overlap exists

**Reply by ramber1836** [reply]
> @ramber1836
> 
> Thanks. But I don't fully understand why the test folder need orders here. My current understanding is, just use the specific one csv from test_x to generate one example's feature, and generate the same name's file in test_y for evaluation. Then loop it.
> 
> If it's ordered, so we can use maybe more test_x files to generate one example? Is this the purpose of test folder ordering?
> 

We try to ensure that different kinds of machine learning methods can be applied, not only multivariate time-series methods but also other types of learning paradigms, such as active learning. 

> Actually, the order also adds the risk of leaking here. If it's ordered, maybe we can use the future test_x to update the previous prediction if the overlap exists

It's true. So the sampling window rolls the test data with random strides and the test_ys cannot be accessed in advance during evaluation.
